pkgs/tools/misc/fwup/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, lib, fetchFromGitHub, autoreconfHook, makeWrapper, pkgconfig
+, zlib, lzma, bzip2, mtools, dosfstools, zip, unzip, libconfuse, libsodium
+, libarchive, darwin, coreutils }:
+
+stdenv.mkDerivation rec {
+  name = "fwup-${version}";
+  version = "0.14.2";
+
+  src = fetchFromGitHub {
+    owner = "fhunleth";
+    repo = "fwup";
+    rev = "v${version}";
+    sha256 = "0ddyiprq4qnqpdhh48bivl8c5yrh21p4r99qs0d1rjiwx5h9p21l";
+  };
+
+  doCheck = true;
+
+  nativeBuildInputs = [ pkgconfig autoreconfHook makeWrapper ];
+  buildInputs = [ zlib lzma bzip2 libconfuse libsodium libarchive ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.DiskArbitration
+    ];
+  propagatedBuildInputs = [ zip unzip mtools dosfstools coreutils ];
+
+  # segaults on darwin without
+  NIX_LDFLAGS = lib.optional stdenv.isDarwin "-F/System/Library/Frameworks";
+
+  # stop test runner from switching to BSD utils
+  preCheck = lib.optional stdenv.isDarwin "sed -i tests/common.sh -e 24,37d";
```

Comment:
This should be better a patch. Otherwise it might silently break with an update.